### PR TITLE
refactor: add RPC interceptors for EVM chains 

### PR DESCRIPTION
**Pull request by guanbinrui**
### Middleware

A component borrows from [Koajs](https://koajs.com/) with the ability to alter JSON RPC payloads and responses, to hijack and redirect requests.

### Interceptor

A special kind of middleware, to hijack requests and customize responses.

### Translator

A component to encode RPC payloads and decode responses for some altered EVM chains.

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -264,7 +264,8 @@
         "xlarge",
         "xlink",
         "zerion",
-        "zubin"
+        "zubin",
+        "middlewares"
```
Comment (nuanyang233) [inline cspell.json]: `middleware` is an uncountable noun with no plural form

File: packages/web3-shared/evm/providers/Injected.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import type { RequestArguments } from 'web3-core'
+import type { EthereumProvider } from '@masknet/injected-script'
+import { EIP1193Provider, EthereumMethodType } from '../types'
+
+export default class InjectedSDK implements EIP1193Provider {
```
Comment (guanbinrui) [inline packages/web3-shared/evm/providers/Injected.ts:5]: Remove SDK suffix. Use `InjectedProvider` instead.

File: packages/injected-script/main/utils.ts [hunk 3]
```diff
@@ -99,7 +99,7 @@ export async function handlePromise(id: number, promise: () => any) {
         sendEvent('resolvePromise', id, data)
     } catch (error) {
         // TODO:
-        sendEvent('rejectPromise', id, {})
+        sendEvent('rejectPromise', id, { message: (error as Error | undefined)?.message })
```
Comment (Jack-Works) [inline packages/injected-script/main/utils.ts]:
```suggestion
        let message: string | undefined
        try {
            message = (error as any).message
        } catch {}
        sendEvent('rejectPromise', id, { message })
```

File: packages/injected-script/sdk/index.ts [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-import { onSolanaEvent } from './bridgedSolana'
+import { onEthEvent } from './injectedEthereumProvider'
+import { onCoin98Event } from './injectedCoin98Provider'
+import { onSolanaEvent } from './injectedSolanaProvider'
 import { sendEvent, rejectPromise, resolvePromise } from './utils'
 
-export { bridgedEthereumProvider } from './bridgedEthereum'
-export { bridgedCoin98Provider } from './bridgedCoin98'
-export { bridgedSolanaProvider } from './bridgedSolana'
+export * from '../shared'
```
Comment (Jack-Works) [inline packages/injected-script/sdk/index.ts:7]: please do not export everything. the surface of this package should keep small. only export things you need.

File: packages/mask/src/plugins/EVM/constants.ts [hunk 5]
```diff
@@ -175,48 +176,69 @@ export const PLUGIN_PROVIDERS: Web3Plugin.ProviderDescriptor[] = [
         type: ProviderType.MaskWallet,
         name: 'Mask Network',
         icon: new URL('./assets/maskwallet.png', import.meta.url),
+        enableRequirements: {
+            supportedChainIds: getEnumAsArray(ChainId).map((x) => x.value),
```
Comment (Jack-Works) [inline packages/mask/src/plugins/EVM/constants.ts:190]: I think explicitly listing all chain IDs can prevent accidental fails in the future.

File: packages/mask/src/extension/background-script/EthereumServices/request.ts [hunk 6]
```diff
@@ -1,261 +1,81 @@
...
+import { Squash } from './middleware/Squash'
+import { Nonce } from './middleware/Nonce'
+import { Interceptor } from './middleware/Interceptor'
+import { Popup } from './middleware/Popup'
+import { Translator } from './middleware/Translator'
+import { RecentTransaction } from './middleware/Transaction'
+import { TransactionNotifier } from './middleware/TransactionNotifier'
+import { TransactionWatcher } from './middleware/TransactionWatcher'
+
+use(new Squash())
```
Comment (Jack-Works) [inline packages/mask/src/extension/background-script/EthereumServices/request.ts:17]: Search for `startEffects` in our code base. Following that pattern will make this file support HMR.

File: packages/injected-script/sdk/index.ts (hunk 4 above)
Comment (Jack-Works) [reply packages/injected-script/sdk/index.ts:7]: @guanbinrui

**Pull request closed without merging**